assignment-section-05/classification_model/processing/validation.py

```diff
@@ -0,0 +1,47 @@
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from pydantic import BaseModel, ValidationError
+
+from classification_model.config.core import config
+from classification_model.processing.data_manager import pre_pipeline_preparation
+
+
+def validate_inputs(*, input_data: pd.DataFrame) -> Tuple[pd.DataFrame, Optional[dict]]:
+    """Check model inputs for unprocessable values."""
+
+    pre_processed = pre_pipeline_preparation(dataframe=input_data)
+    validated_data = pre_processed[config.model_config.features].copy()
+    # TODO: check if we want to drop NA
```

Comment:
@solegalli do we want to drop NA values? If so, which ones? If we do a blanket drop, almost every row is lost.

Reply:
I think we handle NA in the pipeline. I don't think we need to drop extra NA in this dataset. Alternatively, let's do that just in the variables that require imputation.

Reply:
OK, skipping NA drop here.